### PR TITLE
Simulation script

### PR DESCRIPTION
**Pull request by hardy-ethan**

### CODE REVIEW

**Comment by Ryan4253** [comment]
Will test once I get back

**Comment by hardy-ethan** [comment]
works on my machine :tm:

**Comment by hardy-ethan** [comment]
anyone else that reads this is welcome to test, you just have to be in the nav_stack folder and run "scripts/preparesim.bash"

File: scripts/preparesim.bash [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    ZEDSDKFILENAME="zedsdkubuntu22.run"
+
+    if ! [[ -f $ZEDSDKFILENAME ]]; then
+        wget https://download.stereolabs.com/zedsdk/4.0/cu121/ubuntu22 -O $ZEDSDKFILENAME
+    fi
+    chmod +x $ZEDSDKFILENAME
+    ./$ZEDSDKFILENAME
+else
+    sed -i '153,164d' simulation/marvin_simulation/urdf/marvin.xacro
```
Comment (hardy-ethan) [inline scripts/preparesim.bash]: this should probably be replaced with regex, or do we think the line numbers are enough?

**Comment by Ryan4253** [comment]
First installation test fails with the zed sdk installed not able to install anything
image

the script seems to be missing these commands:
```
cd /
export USER=umarv
sudo mkdir -p /etc/udev/rules.d/
```

**Comment by hardy-ethan** [comment]
@Ryan4253 added (modified to remove uselsss `cd /` and to make it work for usernames other than `umarv`

**Comment by Ryan4253** [comment]
Discussed all the bugfixes in Discord, script tested and worked on a fresh env.

Requests
1. Currently looks like ```zstd``` is installed along with all the ROS dependencies. I'd want the install to be in the ZED installation with python3-requests since ```zstd``` is a ZED specific dependency.
2. ```udev``` is not required (up to date in fresh env)
3. ```lshw``` isn't part of the environment, so either add installation or just remove the check (personal recommendation)
4. If the check is still in place, add a notification to the user that a Nvidia GPU is detected
5. ```zedsdkubuntu22.run``` seems to be a bit long of a file name. Seeing that people may have to run the .run file multiple times, might be a good idea to keep it brief like ```zedsdk.run```

**Comment by hardy-ethan** [comment]
@Ryan4253 done

**Comment by Ryan4253** [comment]
fully tested, merging

**Comment by Ryan4253** [comment]
thanks as always

**Pull request merged**